core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -1045,6 +1094,27 @@ public void commitTransaction(SessionContext context, List<TableCommit> commits)
   public List<TableIdentifier> listViews(SessionContext context, Namespace namespace) {
     checkNamespaceIsValid(namespace);
 
+    Map<String, String> queryParams = Maps.newHashMap();
+    if (restPaginationEnabled) {
+      List<TableIdentifier> identifiers = Lists.newArrayList();
+      String pageToken = "";
+      queryParams.put("pageSize", restPageSize);
+
+      while (pageToken != null) {
```

Comment:
I think we should not run the full execution in a loop directly. We should lazily load the list, something like https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/dynamodbv2/datamodeling/PaginatedList.html

Reply:
Makes sense, will revise this

Reply:
+1